src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -58,25 +62,40 @@ internal SdkDirectoryWorkloadManifestProvider(string sdkRootPath, string sdkVers
                 }
             }
 
-            string? userManifestsDir = userProfileDir is null ? null : Path.Combine(userProfileDir, "sdk-manifests", _sdkVersionBand.ToString());
-            string dotnetManifestDir = Path.Combine(_sdkRootPath, "sdk-manifests", _sdkVersionBand.ToString());
-            if (userManifestsDir != null && WorkloadFileBasedInstall.IsUserLocal(_sdkRootPath, _sdkVersionBand.ToString()) && Directory.Exists(userManifestsDir))
-            {
-                _manifestDirectories = new[] { userManifestsDir, dotnetManifestDir };
-            }
-            else
+            if (getEnvironmentVariable(EnvironmentVariableNames.WORKLOAD_MANIFEST_IGNORE_DEFAULT_ROOTS) == null)
             {
-                _manifestDirectories = new[] { dotnetManifestDir };
+                string? userManifestsRoot = userProfileDir is null ? null : Path.Combine(userProfileDir, "sdk-manifests");
+                string dotnetManifestRoot = Path.Combine(_sdkRootPath, "sdk-manifests");
+                if (userManifestsRoot != null && WorkloadFileBasedInstall.IsUserLocal(_sdkRootPath, _sdkVersionBand.ToString()) && Directory.Exists(userManifestsRoot))
+                {
+                    _manifestRoots = new[] { userManifestsRoot, dotnetManifestRoot };
+                }
+                else
+                {
+                    _manifestRoots = new[] { dotnetManifestRoot };
+                }
             }
 
             var manifestDirectoryEnvironmentVariable = getEnvironmentVariable(EnvironmentVariableNames.WORKLOAD_MANIFEST_ROOTS);
             if (manifestDirectoryEnvironmentVariable != null)
```

Comment:
If both `WORKLOAD_MANIFEST_ROOTS` and `WORKLOAD_MANIFEST_IGNORE_DEFAULT_ROOTS` the former will win. Should we consider that an invalid configuration, or should it include both roots?